src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/compilation_task.kt

```diff
@@ -170,6 +170,47 @@ internal fun JvmCompilationTask.kaptArgs(
   }
 }
 
+internal fun JvmCompilationTask.kspArgs(
+  plugins: InternalCompilerPlugins
+): CompilationArgs {
+  return CompilationArgs().apply {
+    plugin(plugins.kspSymbolProcessingCommandLine)
+    plugin(plugins.kspSymbolProcessingApi) {
+      flag("-Xallow-no-source-files")
+
+      val values = arrayOf(
+        "apclasspath" to listOf(inputs.processorpathsList.joinToString(File.pathSeparator)),
+        // projectBaseDir shouldn't matter because incremental is disabled
+        "projectBaseDir" to listOf(directories.incrementalData),
+        // Disable incremental mode
+        "incremental" to listOf("false"),
+        // Directory where class files are written to. Files written to this directory are class
+        // files being written directly from the annotation processor, not Kotlinc
+        "classOutputDir" to listOf(directories.generatedClasses),
+        // Directory where generated Java sources files are written to
+        "javaOutputDir" to listOf(directories.generatedJavaSources),
+        // Directory where generated Kotlin sources files are written to
+        "kotlinOutputDir" to listOf(directories.generatedSources),
+        // Directory where META-INF data is written to. This might not be the most ideal place to
+        // write this. Maybe just directly to the classes directory?
+        "resourceOutputDir" to listOf(directories.generatedSources),
+        // TODO(bencodes) Not sure what this directory is yet.
+        "kspOutputDir" to listOf(directories.incrementalData),
+        // Directory to write KSP caches. Shouldn't matter because incremental is disabled
+        "cachesDir" to listOf(directories.incrementalData),
+        // Set withCompilation to false because we run this as part of the standard kotlinc pass
+        "withCompilation" to listOf("false"),
```

Comment:
With `withCompilation=false` wouldn't this require an extra compilation pass to compile the generated ksp code?

Reply:
The outputs get swept up by the existing Kotlinc pass. There are some tests in place that assert that the compiled class files exist. https://github.com/bazelbuild/rules_kotlin/pull/679/files#diff-bdab9cf068d9747c4ade4192a1d414ebbe1e33d1634b6747fa6e3b942899e1e4R108